exdb/forms.py

```diff
@@ -73,59 +75,62 @@ class Meta:
 
     def __init__(self, *args, **kwargs):
         when = kwargs.pop('when', now())
+        submit = kwargs.pop('submit', None)
         super(ExperienceSaveForm, self).__init__(*args, **kwargs)
         self.when = when
+        self.approval_form = submit
 
 
 class ExperienceSubmitForm(ExperienceSaveForm):
 
     def clean(self):
-
-        if not self.cleaned_data.get('description'):
-            raise ValidationError(_('A description is required'))
-
-        if not self.cleaned_data.get('end_datetime'):
-            raise ValidationError(_('An end time is required'))
-
-        if not self.cleaned_data.get('start_datetime'):
-            raise ValidationError(_('A start time is required'))
-
-        if not self.cleaned_data.get('sub_type'):
-            raise ValidationError(_('The sub type field is required'))
-
-        if not self.cleaned_data.get('type'):
-            raise ValidationError(_('The type field is required'))
-
-        ex_type = self.cleaned_data.get('type')
-
-        if not self.cleaned_data.get('next_approver') and ex_type.needs_verification:
-            raise ValidationError(_('Please select the supervisor to review this experience'))
-
-        if self.cleaned_data.get('start_datetime') >= self.cleaned_data.get('end_datetime'):
-            raise ValidationError(_('Start time must be before end time'))
-
-        if not ex_type.needs_verification and self.cleaned_data.get('start_datetime') > self.when:
-            raise ValidationError(_('%(name)s experiences must have happened in the past') % {'name': ex_type.name})
-
-        if ex_type.needs_verification and self.cleaned_data.get('start_datetime') < self.when:
-            raise ValidationError(_('%(name)s events cannot happen in the past') % {'name': ex_type.name})
-
-        if not ex_type.needs_verification and (not self.cleaned_data.get(
-                'attendance') or self.cleaned_data.get('attendance') < 1):
-            raise ValidationError(_('%(name)s events must have an attendance') % {'name': ex_type.name})
-
-        if not ex_type.needs_verification and not self.cleaned_data.get('audience'):
-            raise ValidationError(_('%(name)s events must have an audience') % {'name': ex_type.name})
-
-        if ex_type.needs_verification and self.cleaned_data.get('attendance'):
-            raise ValidationError(_('%(name)s events cannot have an attendance') % {'name': ex_type.name})
-
-        # There are too many branches in this function; this is fixed on
-        # the approval validation branch.
-        if not ex_type.needs_verification and not self.cleaned_data.get('conclusion'):
-            raise ValidationError(_('%(name)s events must have a conclusion') % {'name': ex_type.name})
-
-        if ex_type.needs_verification and self.cleaned_data.get('conclusion'):
+        needs_verification = self.cleaned_data.get(
+            'type').needs_verification if self.cleaned_data.get('type') else False
```

Comment:
`(self.cleaned_data.get('type') and not needs_verification)` and `(self.cleaned_data.get('type') and needs_verification)` is repeated many times in the code below. Perhaps `needs_verification` should be a trinary variable (`True`/`False`/`None`) so the rest of the code is simpler.

For example:
```python
needs_verification = self.cleaned_data.get('type').needs_verification if self.cleaned_data.get('type') else False
...
(self.cleaned_data.get('type') and not needs_verification)
```
changes to:
```python
type = self.cleaned_data.get('type')
needs_verification = type.needs_verification if type else None
...
(needs_verification is False)
```